test/extended/util/prometheus/helpers.go

```diff
@@ -45,13 +46,50 @@ type prometheusResponseData struct {
 }
 
 // GetBearerTokenURL makes http request with bearer token
-func GetBearerTokenURL(url, bearer string) (string, error) {
-	cmd := fmt.Sprintf("curl --retry 15 --max-time 2 --retry-delay 1 -s -k -H 'Authorization: Bearer %s' %q", bearer, url)
-	output, err := exec.Command("bash", "-e", "-c", cmd).Output()
+func GetBearerTokenURL(url, bearerToken string) (string, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{
+		Timeout:   time.Duration(10 * time.Second),
```

Comment:
Not sure we need this timeout if we retry with a timeout below?

Reply:
`PollUntilContextTimeout` execute the condition function after every interval has passed, until the timeout. I put the same timeout for the HTTP client as the interval of calling the condition function. So that by the time the second HTTP request is sent, the first has timed out already.